### DISCUSSION TITLE
HTTPS and LDAP

### DISCUSSION THREAD

**Discussion by JeremyGarman** [open-webui/open-webui#2993] [General]
Two quick questions:
1.  Is there an easy way to incorporate an internal CA-signed cert such that my URL is actually https?  (corporate security requirement)  I'm trying to deploy Open-webui using the manual method or a RHEL 8 server  (rather than Ubuntu 20.04 or 22.04 like everyone else seems to be doing)
2.  Rather than using the internal authentication method, can Open-webui point to an LDAP source (say a corporate Active Directory instance) for RBAC compliance (again, corporate security requirement)

Thanks in advance!

**Comment by frankkubler** [comment]
Hello, 

for https the easiest way is to add at the en of the file :  /backend/start.sh where it is the uvicorn is definded, the path to ssl cert and key file.
Mine lookss like that : 

`WEBUI_SECRET_KEY="$WEBUI_SECRET_KEY" exec uvicorn main:app --host "$HOST" --port "$PORT" --forwarded-allow-ips '*' --ssl-certfile="/app/certs/live/mydomain.duckdns.org/cert.pem" --ssl-keyfile="/app/certs/live/mydomain.duckdns.org/privkey.pem"  `

it works like a charm

for LDAP i'm interessted too but I have no idea.

**Reply by JeremyGarman** [reply]
Thanks very much, I'll give that a try!

**Comment by JuanjoJG-WATA** [comment]
What @kubikub says is correct, that's how I did it too. I also changed to port to 443.

**Reply by JeremyGarman** [reply]
That's exactly my intention as well!  You simply defined port 443 in the "--port" option in start.sh ?

**Reply by JuanjoJmnz** [reply]
That's right! Make sure you are always in sudo or you may find a few problems the same way I did.

[I explained it here, just in case](https://github.com/open-webui/open-webui/discussions/2879)

And yeah, I'm the same guy, I can't login on my main acc from my phone 😂

**Comment by frankkubler** [comment]
I'm using open-webui in a docker so, i did not change port, I used the default port 3000(docker configuration) and on my internet box or server, I redirected port 13000 to 3000. After what I can connect open-webui with https://mydomain.duckdns.org:13000

However, I did not found yet how I can change start.sh options in the docker-compose.yaml. For the moment I did it manually with docker exec , explore the docker volume and nano start.sh. So each update I have to do that.

**Reply by JuanjoJG-WATA** [reply]
Yeah, that's why I decided to install it the manual way, using Node.js and Python. It let me do some changes that i couldn't/didn't know how to do them with Docker.

**Reply by JeremyGarman** [reply]
So speaking of manual install...   :)
What OS did you use for your base?  For security governance reasons I need to use hardened RHEL 8, so I always end up having differences than what all the predominantly Ubuntu (20.04 LTS or 22.04 LTS) installations describe.  I'm assuming I'll need a virtualenv with Python 3.11 installed there (so as not to conflict with the native Python 3.6 or 3.7 in RHEL 8 OS), and then install node.js when the virtenv is active, do the ollama install there and then pull the llama3 model that I want to use.

Like what you just described, I am great working in the OS, but I have very little experience with containers in general and docker in particular, so I'd rather run it using apache httpd or tomcat... but then there's still the LDAP integration part that's up in the air...   :)
